Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#30917 Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN**

This is a C++14 feature, we can use this now.

Differential Revision: [D18869637](https://our.internmc.facebook.com/intern/diff/D18869637/)

smessmer committed:
Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN

This is a C++14 feature, we can use this now.

Differential Revision: [D18869637](https://our.internmc.facebook.com/intern/diff/D18869637/)

[ghstack-poisoned]

ezyang reviewed (approved):
Meta-note: it's probably better for you if you don't stack these; I bet at least *some* of these changes are going to run afoul portability problems with other platforms we support...

smessmer committed:
Update on "Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN"


This is a C++14 feature, we can use this now.

Differential Revision: [D18869637](https://our.internmc.facebook.com/intern/diff/D18869637/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN"


This is a C++14 feature, we can use this now.

Differential Revision: [D18869637](https://our.internmc.facebook.com/intern/diff/D18869637/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN"


This is a C++14 feature, we can use this now.

Differential Revision: [D18869637](https://our.internmc.facebook.com/intern/diff/D18869637/)

[ghstack-poisoned]

smessmer committed:
Update on "Use [[noreturn]] instead of C10_NORETURN or CAFFE_NORETURN"


This is a C++14 feature, we can use this now.

Differential Revision: [D18869637](https://our.internmc.facebook.com/intern/diff/D18869637/)

[ghstack-poisoned]
